File: 01431-kids-with-the-greatest-number-of-candies/Solution.php
<?php

declare(strict_types=1);

class Solution {

    /**
     * @param array<int> $candies
     * @param int $extraCandies
     * @return array<bool>
     */
    function kidsWithCandies(array $candies, int $extraCandies): array {      
        $max = max($candies);
        return array_map(fn($kidCandies) => ($kidCandies + $extraCandies) >= $max, $candies);
    }
}

// 0ms beats 100.00% of users with PHP, 18.26MB beats 100.00% of users with PHP

File: 01431-kids-with-the-greatest-number-of-candies/solution.go
func kidsWithCandies(candies []int, extraCandies int) []bool {
	var maxCandies int
	for _, n := range candies {
		if maxCandies < n {
			maxCandies = n
		}
	}

	var res []bool
	for _, n := range candies {
		if (n + extraCandies) >= maxCandies {
			res = append(res, true)
			continue
		}
		res = append(res, false)
	}

	return res
}

// 2ms Beats 76.45% of users with Go, 2.40MB Beats 93.82% of users with Go